module/zfs/spa.c

```diff
@@ -702,6 +708,24 @@ spa_prop_validate(spa_t *spa, nvlist_t *props)
 				error = SET_ERROR(E2BIG);
 			break;
 
+		case ZPOOL_PROP_FEATURES:
+			/*
+			 * can't do a full check of the contents of
+			 * the features file here, so basic sanity check -
+			 * is it a plausible pathname?
+			 */
+			if ((error = nvpair_value_string(elem, &strval)) != 0)
+				break;
+			for (check = strval; *check != '\0'; check++) {
+				if (!isprint(*check)) {
```

Comment:
Given that `isprint` takes a `char`, I think it only understands ASCII, but a valid filename could contain unicode characters (`features=🎉`, anyone?).  I think if you want to do a check we would have to see if the file actually exists (which I think the kernel can do?).

Reply:
I don't think we *want* to be throwing any errors here; I'm not even sure whether the current check is warranted. This property should only affect the operation of `zpool create`, `zpool upgrade` and (slightly) `zpool status`; it should otherwise have no effect. My thinking here is that it's probably a good idea to test for an obviously nonsensical value, but otherwise to leave any/all parsing of the property to the zpool binary. The current check is a straightforward copy of the `ZPOOL_PROP_COMMENT` check immediately above.

WDYT?

Reply:
ACK; I think that removing this check completely is sensible.